Fix GCC 15 compatibility in bcl2fastq2 and its bundled Boost:
- use the Boost source pre-extracted by EasyBuild so it can be patched directly;
- add the missing filesys.h and cstdint headers;
- correct the pool_ member name in StaticMemPool copy operations.
Author: Emik Lin (HKUMed CPOS)
--- a/src/cmake/bootstrap/installBoost.sh
+++ b/src/cmake/bootstrap/installBoost.sh
@@ -32,5 +32,4 @@
 SCRIPT=`basename "$0"`
 VERSION=`echo ${BCL2FASTQ_BOOST_VERSION} | sed "s/\./_/g"`
-SOURCE_TARBALL=${REDIST_DIR}/boost_${VERSION}.tar.bz2
-TARBALL_COMPRESSION=j
-SOURCE_DIR=${BUILD_DIR}/boost_${VERSION}
+# Boost is pre-extracted so EasyBuild can patch it before configuration.
+SOURCE_DIR=$(readlink -f "${REDIST_DIR}/../boost_${VERSION}")
@@ -40,5 +39,4 @@
 if [[ $CLEAN ]] ; then
-    echo removing $SOURCE_DIR
-    rm -rf $SOURCE_DIR ${INCLUDE_DIR}/boost ${LIB_DIR}/libboost_*.{a,so}
+    rm -rf ${INCLUDE_DIR}/boost ${LIB_DIR}/libboost_*.{a,so}
     exit 0
 fi
@@ -46,2 +44,1 @@
-common_create_source
 cd ${SOURCE_DIR} \
--- a/src/cxx/include/common/StaticMemPool.hpp
+++ b/src/cxx/include/common/StaticMemPool.hpp
@@ -46,6 +46,6 @@
 : pool_()
 , metadata_()
 {
-    std::copy(&that.pool_[0][0], &that.pool[PageCount-1][PageSize], &pool_[0][0]);
+    std::copy(&that.pool_[0][0], &that.pool_[PageCount-1][PageSize], &pool_[0][0]);
     std::copy(&that.metadata_[0], &that.metadata_[PageCount], &metadata_[0]);
 }
@@ -55,7 +55,7 @@
 {
     if (this != &rhs)
     {
-        std::copy(&rhs.pool_[0][0], &rhs.pool[PageCount-1][PageSize], &pool_[0][0]);
+        std::copy(&rhs.pool_[0][0], &rhs.pool_[PageCount-1][PageSize], &pool_[0][0]);
         std::copy(&rhs.metadata_[0], &rhs.metadata_[PageCount], &metadata_[0]);
     }
     return *this;
--- a/boost_1_54_0/tools/build/v2/engine/modules/path.c
+++ b/boost_1_54_0/tools/build/v2/engine/modules/path.c
@@ -9,3 +9,4 @@
 #include "../lists.h"
 #include "../native.h"
+#include "../filesys.h"
 #include "../timestamp.h"
--- a/boost_1_54_0/boost/atomic/atomic.hpp
+++ b/boost_1_54_0/boost/atomic/atomic.hpp
@@ -11,2 +11,3 @@
 #include <cstddef>
+#include <cstdint>
 #include <boost/cstdint.hpp>
